src/cache_memoize/__init__.py

```diff
@@ -1,10 +1,11 @@
+import hashlib
 from functools import wraps
 
-import hashlib
 from django.core.cache import caches, DEFAULT_CACHE_ALIAS
-
 from django.utils.encoding import force_text, force_bytes
 
+NONE_PLACEHOLDER = '___NONE_PLACEHOLDER___'
```

Comment:
Instead of an odd string, can't this be `NONE_PLACEHOLDER = object()`?

Reply:
Also, if you use `object()` instead, let's call it simply `MARKER` instead of `NONE_PLACEHOLDER` since that lingo is common for patterns like this.

Reply:
I am sorry, I don't see how it can work with `object()`.
Please consider this code:
```

from django.core.cache import cache

MARKER = object()

cache.set('marker', MARKER)

print(id(MARKER))
# 140275214923040

marker = cache.get('marker')

print(id(marker))
# 140275214923056

print(marker == MARKER)
# False

print(id(marker) == id(MARKER))
# False

print(marker is MARKER)
# False
```
The point of `NONE_PLACEHOLDER` to cache value which will mean `None` but won't be `None`. I can't see how we can achieve this with `object()`.

Reply:
The trick is to never do `cache.set('mykey', MARKER)`. 

The core of the idea is this:

```python
marker = object()
value = cache.get('key', marker)
if value is marker:
    # the key was not in the cache
    value = compute_value()
    cache.set('key', value)
return value
```

If `cache` was a pure Python dict you wouldn't need this because you could do:
```python
if 'key' in cache_dict:
    value = cache_dict['key']
else:
    value = compute_value()
    cache_dict['key'] = value
return value
```

So that's, potentially, **two read queries** from the data structure. One to ask "Do you even have this key?" and one to ask "Alright, what's the value of this key?". If the network latency matters, it's oftentimes more performant to send only **1 read query** and deal with consequences if it comes back empty. 

But, even with a pure Python dict, it's sometimes pleasant to use the `.get()` method because you only need *1* branch:

```python
marker = object()
value = cache_dict.get('key', marker)
if value is marker:
    # the key was not in the cache dict
    value = compute_value()
    cache_dict['key'] = value
return value
```

In the last example, suppose you didn't use the "marker techique"...


```python
value = cache_dict.get('key')
if value is None:
    # the key was PROBABLY not in the cache dict
    value = compute_value()
    cache_dict['key'] = value
return value
```

...then, what if `compute_value()` genuinely returns None. What *would* happen is that the caching would not protect `compute_value()`. And that's what you want to achieve.

Reply:
Makes sense, that'll definitely make code look better. Will update PR tomorrow.